sympy/printing/lambdarepr.py

```diff
@@ -336,6 +336,31 @@ def _print_Function(self, e):
                                 func_name)
         return "%s(%s)" % (nstr, self._print_seq(e.args))
 
+    def _print_Piecewise(self, expr):
+        "Piecewise function printer"
+        exprs = [self._print(arg.expr) for arg in expr.args]
+        conds = [self._print(arg.cond) for arg in expr.args]
+        # If [default_value, True] is a (expr, cond) sequence in a Piecewise object
+        #     it will behave the same as passing the 'default' kwarg to select()
+        #     *as long as* it is the last element in expr.args.
+        # If this is not the case, it may be triggered prematurely.
+        ans = []
+        parenthesis_count = 0
+        is_last_cond_True = False
+        for cond, expr in zip(conds, exprs):
+            if cond == 'True':
+                ans.append(expr)
+                is_last_cond_True = True
+                break
+            else:
+                ans.append('where(%s,%s,' % (cond, expr))
+                parenthesis_count += 1
+        if not is_last_cond_True:
+            # simplest way to put a nan but raises
+            # 'RuntimeWarning: invalid value encountered in log'
+            ans.append('log(-1)')
```

Comment:
Sounds good, doesn't work.

```
>>> numexpr.evaluate("float('nan')")
TypeError: 'VariableNode' object is not callable
```

If anybody has other ideas, you can test them with `numexpr.evaluate`.

I think the only solution is to assign float('nan') or numpy.nan to a variable and use it in the expression, but the variable name could conflict with the variables of the expression. Maybe we should try names of the form 'nan%s' % bigrandomnumer until one is free but I'm not comfortable enough with the numexpr and sympy namespaces handling system to make a flawless code.